agent/agent.go

```diff
@@ -0,0 +1,218 @@
+package agent
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/orbs-network/scribe/log"
+)
+
+const DDMMYYYYhhmmss = "2006-01-02 15:04:05"
+
+type Config struct {
+	IntervalMinute uint
+	Url            string
+}
+
+type Agent struct {
+	config *Config
+	ticker *time.Ticker
+}
+
+var single *Agent
+var logger log.Logger
+
+func Init(c *Config) {
+	//initialize static instance on load
+	single = &Agent{config: c}
+}
+
+//GetInstanceA - get singleton instance pre-initialized
+func GetInstance() *Agent {
+	return single
+}
+
+/////////////////////////////
+func (a *Agent) Start(start bool) {
+	if start {
+		if a.ticker == nil {
+			dlPath := getDownloadPath()
+
+			// ensure download hash folder
+			err := os.MkdirAll(dlPath, 0777)
+			if err != nil {
+				log.Error(err)
+			}
+
+			logger.Info("start boyar Agent")
+			tick(a.config.Url, dlPath)
+			a.ticker = time.NewTicker(5 * time.Second) // DEBUG
```

Comment:
Let's remove the debug code from here please. Make the first tick always happen when Boyar starts so there's no need for a debug flow. Then we can run with the real timeouts. These things are dangerous because people will forget them in the production code and deploy like this by accident.